Introduce the new "shared credentials" format (#497)
apple/password-manager-resources: .github/PULL_REQUEST_TEMPLATE.md

- [ ] The top-level JSON objects are sorted alphabetically
- [ ] There are no [open pull requests](https://github.com/apple/password-manager-resources/pulls) for the same update

> #### for websites-with-shared-credential-backends.json

Reviewer: I think we should also make an update to the top-level `README.md` file so contributors won't continue to be update `websites-with-shared-credential-backends.json`. We can remove mentions of this file except maybe calling out it's deprecated and generated for legacy reasons in favor of these new files you've added.

- [ ] The given rule isn't particularly standard and obvious for password managers
- [ ] Generated passwords have been tested from this rule using the [Password Rules Validation Tool](https://developer.apple.com/password-rules/)
- [ ] Information has been included about the website's requirements (eg. screenshots, error messages, steps during experimentation, etc.)
- [ ] The PR isn't documenting something that would be a common practice among password managers (e.g. minimal length of 6)

#### for change-password-URLs.json
- [ ] There is no Well-Known URL for Changing Passwords (`https://example.com/.well-known/change-password`)

> - [ ] The URL either makes the experience better or no worse than being directed to just the domain in a non-logged-in state

Reviewer 2: Nit pick: if this section had stayed where it was, the sections would be alphabetized by filename. That's kinda nice.

Author: I made an intentional change here to start ordering these files by how often they’re changed, with the most common one up front.

> #### for shared-credential.json

Reviewer: Small typo as the name of the actual file is plural
```suggestion
#### for shared-credentials.json
```

- [ ] There's evidence the domains are currently related (SSL certificates, DNS entries, valid links between sites, legal documents etc.)

> - [ ] The new group serves login pages on each of the included domains, and those login page accept accounts from the others. (For example, we don't associate `google.co.il` to `google.com`, because `google.co.il` redirects to `accounts.google.com` for authentication.)

Reviewer 3: We should probably mention redirects here as well :slightly_smiling_face:

Reviewer 2: @Cldfire what do you mean by redirects?

Reviewer 3: @erynofwales one of the main motivations behind the updated format is first-class support for declaring redirects 🙂 such as `discordapp.com` -> `discord.com`.

The concept generalizes beyond that, but explaining it terms of its most common form (redirects) feels like a good fit for the pull request template.

> - [ ] The new group serves login pages on each of the included domains, and those login page accept accounts from the others. (For example, we don't associate `google.co.il` to `google.com`, because `google.co.il` redirects to `accounts.google.com` for authentication.)

Reviewer 2: Spelling nit: "and those login page" -> "and those login pages"

> #### for shared-credential-historical.json

Reviewer: Small typo with filename here as well
```suggestion
#### for shared-credentials-historical.json
```

> - [ ] You believe that the domains were ever associated and can explain that relationship

Reviewer 3: May be a bit easier to read written as:

```suggestion
- [ ] You believe that the domains were associated at some point in the past and can explain that relationship
```
